### PR TITLE
feat: added an option to prefer silk touch with the tool switch

### PR DESCRIPTION
**Pull request by meza**
Been wanting this for the longest time.

I've added a relatively naive implementation for now. Wasn't sure if you had a better way of comparing a targeted block with the names in the list so please feel free to point me to a "tweakeroo" way of handling it :)

We could probably complicate the tool selection logic even further in the case there's multiple eligible tools with silk touch but I'm not convinced that would be necessary

### CODE REVIEW

File: src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java [hunk 1]
```diff
@@ -38,13 +41,13 @@
...
-import fi.dy.masa.tweakeroo.config.Configs;
-import fi.dy.masa.tweakeroo.config.FeatureToggle;
-import fi.dy.masa.tweakeroo.tweaks.PlacementTweaks;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
Comment (meza) [inline src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java:50]: Eh, I just realised my "optimise imports" messed this up. Do you want me to fix it?

File: src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java [hunk 2]
```diff
@@ -581,6 +599,12 @@ else if (getBaseBlockBreakingSpeed(testedStack, state) == getBaseBlockBreakingSp
         return false;
     }
 
+    private static boolean toolHasSilkTouch(ItemStack stack)
+    {
+        // 1.21 way of checking for Silk Touch according to the Fabric upgrade blog post.
```
Comment (meza) [inline src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java]: https://fabricmc.net/2024/05/31/121.html#Enchantments:~:text=EnchantmentHelper%23hasAnyEnchantmentsIn(stack%2C%20EnchantmentTags%23PREVENTS_BEE_SPAWNS_WHEN_MINING)

Comment (sakura-ryoko) [reply src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java]: There are better ways to do this, such as using 'getEnchantmentLevel()` from my code.

Comment (meza) [reply src/main/java/fi/dy/masa/tweakeroo/util/InventoryUtils.java]: That's what I started with and implemented for 1.20.6... Then I realised I was on the 1.20.6 branch and switched to 1.21 :D Apparently checking for a specific enchantment is not as clean anymore, they're trying to get away from enchantment names over to their effects without actually listing all the possible effects -.-'

**Comment by sakura-ryoko** [comment]
Will be for next release.

**Pull request merged**